test/tokenizer_test.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:pana/src/license_detection/tokenizer.dart';
+import 'package:test/scaffolding.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('Tokenizer tests', () {
+    test('Ignore pure punctuations', () {
+      final text = '// hello! ^& world %^& 1.1';
+      final expected = <String>['hello', 'world', '1.1'];
+      final actual = tokenize(text);
+
+      testOutput(actual, expected);
+    });
```

Comment:
These tests are really nice!!! 

Consider making a helper function like:
```dart
testTokenizer(String name, {required String text, required List<String> tokens}) => test(name, () {
  testOutput(tokenize(text), tokens);
});

// Then you can just write each test case as:
testTokenize(
  'Ignore pure punctuations',
  text: '// hello! ^& world %^& 1.1',
  tokens: ['hello', 'world', '1.1'],
);
```

Reply:
Another option is to make a `Map<String,List<String>>` with test vectors and just use the input text as the name of the test case..

This is only really to make easier to write a lot of small test cases. Which is probably a good idea to do :D